rbd/rbd_pacific.go

```diff
@@ -0,0 +1,79 @@
+// +build !octopus,!nautilus
+
+package rbd
+
+// #cgo LDFLAGS: -lrbd
+// /* force XSI-complaint strerror_r() */
+// #define _POSIX_C_SOURCE 200112L
+// #undef _GNU_SOURCE
+// #include <errno.h>
+// #include <stdlib.h>
+// #include <rados/librados.h>
+// #include <rbd/librbd.h>
+import "C"
+
+import (
+	"unsafe"
+)
+
+const (
+	RBD_ENCRYPTION_FORMAT_LUKS1 = 0
+	RBD_ENCRYPTION_FORMAT_LUKS2 = 1
+)
+
+const (
+	RBD_ENCRYPTION_ALGORITHM_AES128 = 0
+	RBD_ENCRYPTION_ALGORITHM_AES256 = 1
+)
+
+type EncryptionFormatOptsLUKS1 struct {
+	Alg            uint
+	Passphrase     string
+	PassphraseSize uint
+}
+
+type EncryptionFormatOptsLUKS2 struct {
+	Alg            uint
+	Passphrase     string
+	PassphraseSize uint
+}
+
+// Create an encryption format header
+//
+// Implements:
+//  int rbd_encryption_format(rbd_image_t image,
+//                            rbd_encryption_format_t format,
+//                            rbd_encryption_options_t opts,
+//                            size_t opts_size);
+func (image *Image) EncryptionFormat(format uint, opts unsafe.Pointer) error {
```

Comment:
`unsafe.Pointer` should probably not be passed by a user calling this function. Please consider using the go-types only for public functions.